File: design-patterns/go/factory/transport-method/transport_factory.go
// transport_factory.go

package main

type TransportFactoryInterface interface {
	GetTransport(name string) (Transport, error)
	OperateTransport(name string)
	RepairTransport(name string)
}

type TransportFactory struct {
	TransportFactoryInterface
}

func NewTransportFactory() (transportFactory *TransportFactory) {
	transportFactory = &TransportFactory{}
	return
}

func (transportFactory *TransportFactory) OperateTransport(name string) {
	transport, err := transportFactory.GetTransport(name)

	if err == nil {
		transport.Start()
		transport.Stop()
	}
}

func (transportFactory *TransportFactory) RepairTransport(name string) {
	transport, err := transportFactory.GetTransport(name)

	if err == nil {
		transport.Repair()
	}
}

File: design-patterns/php/factory/transport-method/TransportFactory.php
<?php
// TransportFactory.php

namespace Factory\TransportMethod;

abstract class TransportFactory {
    public function operateTransport(string $name) {
        $transport = $this->getTransport($name);

        $transport->start();
        $transport->stop();
    }

    public function repairTransport(string $name) {
        $transport = $this->getTransport($name);

        $transport->repair();
    }

    public abstract function getTransport(string $name): ?Transport;
}